test/tla/ERC20_typedefs.tla

```diff
@@ -0,0 +1,26 @@
+------------------------ MODULE ERC20_typedefs --------------------------------
+(*
+  Type definitions for the module ERC20.
+
+  An account address, in our case, simply an uninterpreted type ADDR.
+
+  @typeAlias: transfer = { id: Int, fail: Bool, sender: ADDR, toAddr: ADDR, value: Int };
+  @typeAlias: approve = { id: Int, fail: Bool, sender: ADDR, spender: ADDR, value: Int };
+  @typeAlias: transferFrom = { id: Int, fail: Bool, sender: ADDR, fromAddr: ADDR, toAddr: ADDR, value: Int };
+
+  A transaction is a discriminated union:
+  @typeAlias: TX = None({ id: Int, fail: Bool }) | Transfer($transfer) | Approve($approve) | TransferFrom($transferFrom) ;
```

Comment:
How could `None` fail?

Reply:
The value is unused, as is the `id`. I removed them.